src/main/java/io/openapitools/swagger/GenerateMojo.java

```diff
@@ -152,14 +156,34 @@ private Application resolveApplication(JaxRSScanner reflectiveScanner) {
         Class<? extends Application> appClazz = (Class<? extends Application>)clazz;
         return ClassUtils.createInstance(appClazz);
     }
-    
+
     private URLClassLoader createClassLoader() {
         try {
-            File compiled = new File(project.getBuild().getOutputDirectory());
-            return new URLClassLoader(new URL[] {compiled.toURI().toURL()}, Thread.currentThread().getContextClassLoader());
+            Collection<String> dependencies = getDependentClasspathElements();
```

Comment:
I am thinking the conversion can be done something like:

```java
URL[] urls = getDependentClasspathElements().stream()
  .map(dependency -> Paths.get(dependency).toUri().toURL())
  .toArray(URL[]::new)
```

Reply:
Unfortunately, it is not possible with such a simple looking stream transformation (which I would also favor), because `URI#toURL()` throws `MalformedURLException`, which is a checked exception, and streams do not like them (probably nobody likes them :)). It would have to be changed to something like this:
```
URL[] urls = getDependentClasspathElements().stream()
  .map(dependency -> {
    try {
      return Paths.get(dependency).toUri().toURL());
    } catch (MalformedURLException e) {
      throw new RuntimeException("Some exception", e);
    }
  }
  .toArray(URL[]::new);
```
So the conciseness of the stream transformations is pretty much gone. But it is not a problem at all to change it to the above code if you would like to, so it is your call.

Reply:
bugger those checked exceptions still linger on... Its a good point... Lets keep it as is...